android/java/org/chromium/chrome/browser/crypto_wallet/util/WalletUtils.java

```diff
@@ -106,16 +107,31 @@ public static AccountInfo getAccountInfoFromIntent(@NonNull final Intent intent)
         return AccountInfo.deserialize(ByteBuffer.wrap(bytes));
     }
 
-    public static void openWebWallet() {
+    public static void openWebWallet(final boolean forceNewTab) {
         try {
             BraveActivity activity = BraveActivity.getBraveActivity();
-            activity.openNewOrSelectExistingTab(BraveActivity.BRAVE_WALLET_URL, true);
+            if (forceNewTab) {
+                activity.openAlwaysNewAndCloseExistingTab(
+                        BraveActivity.BRAVE_WALLET_BASE_URL, BraveActivity.BRAVE_WALLET_URL);
```

Comment:
I would suggest to implement that as  `if (forceNewTab) {closeAllTabsByOrigin(...)`} followed by `openNewOrRefreshExistingTab(...)`